exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1333 @@
+		return by.UpdateTicker(ctx, p, assetType)
+	}
+	return tickerNew, nil
+}
+
+// FetchOrderbook returns orderbook base on the currency pair
+func (by *Bybit) FetchOrderbook(ctx context.Context, currency currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	ob, err := orderbook.Get(by.Name, currency, assetType)
+	if err != nil {
+		return by.UpdateOrderbook(ctx, currency, assetType)
+	}
+	return ob, nil
+}
+
+// UpdateOrderbook updates and returns the orderbook for a currency pair
+func (by *Bybit) UpdateOrderbook(ctx context.Context, p currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	var orderbookNew Orderbook
+	var err error
+
+	formattedPair, err := by.FormatExchangeCurrency(p, assetType)
+	if err != nil {
+		return nil, err
+	}
+
+	book := &orderbook.Base{
+		Exchange:        by.Name,
+		Pair:            formattedPair,
+		Asset:           assetType,
+		VerifyOrderbook: by.CanVerifyOrderbook,
+	}
+
+	switch assetType {
+	case asset.Spot:
+		orderbookNew, err = by.GetOrderBook(ctx, formattedPair.String(), 0)
+	case asset.CoinMarginedFutures, asset.USDTMarginedFutures, asset.Futures:
+		orderbookNew, err = by.GetFuturesOrderbook(ctx, formattedPair)
+	default:
+		return nil, fmt.Errorf("assetType not supported: %v", assetType)
+	}
+	if err != nil {
+		return book, err
+	}
+
+	for x := range orderbookNew.Bids {
+		book.Bids = append(book.Bids, orderbook.Item{
+			Amount: orderbookNew.Bids[x].Amount,
+			Price:  orderbookNew.Bids[x].Price,
+		})
+	}
+
+	for x := range orderbookNew.Asks {
+		book.Asks = append(book.Asks, orderbook.Item{
+			Amount: orderbookNew.Asks[x].Amount,
+			Price:  orderbookNew.Asks[x].Price,
+		})
+	}
+	err = book.Process()
+	if err != nil {
+		return book, err
+	}
+	return orderbook.Get(by.Name, formattedPair, assetType)
+}
+
+// UpdateAccountInfo retrieves balances for all enabled currencies
+func (by *Bybit) UpdateAccountInfo(ctx context.Context, assetType asset.Item) (account.Holdings, error) {
+	var info account.Holdings
+	var acc account.SubAccount
+	info.Exchange = by.Name
+	switch assetType {
+	case asset.Spot:
+		balances, err := by.GetWalletBalance(ctx)
+		if err != nil {
+			return info, err
+		}
+
+		currencyBalance := make([]account.Balance, len(balances))
+		for i := range balances {
+			currencyBalance[i] = account.Balance{
+				CurrencyName: currency.NewCode(balances[i].CoinName),
+				Total:        balances[i].Total,
+				Hold:         balances[i].Locked,
+			}
+		}
+
+		acc.Currencies = currencyBalance
+
+	case asset.CoinMarginedFutures, asset.USDTMarginedFutures, asset.Futures:
+		balances, err := by.GetFutureWalletBalance(ctx, "")
+		if err != nil {
+			return info, err
+		}
+
+		var i int
+		currencyBalance := make([]account.Balance, len(balances))
+		for coinName, data := range balances {
+			currencyBalance[i] = account.Balance{
+				CurrencyName: currency.NewCode(coinName),
+				Total:        data.Equity,
+				Hold:         data.Equity - data.AvailableBalance,
```

Comment:
These can now be updated to hold more data. Account.Balance now has the following fields:
```
	Total                  float64
	Hold                   float64
	Free                   float64
	AvailableWithoutBorrow float64
	Borrowed               float64
```